Libraries/LibWeb/TrustedTypes/TrustedTypePolicy.cpp

```diff
@@ -238,4 +238,78 @@ WebIDL::ExceptionOr<JS::Value> process_value_with_a_default_policy(TrustedTypeNa
     }
 }
 
+// https://www.w3.org/TR/trusted-types/#get-trusted-type-compliant-string-algorithm
+WebIDL::ExceptionOr<Utf16String> get_trusted_type_compliant_string(TrustedTypeName expected_type, JS::Object& global, Variant<GC::Root<TrustedHTML>, GC::Root<TrustedScript>, GC::Root<TrustedScriptURL>, Utf16String> input, InjectionSink sink, String sink_group)
+{
+    auto& vm = global.vm();
+
+    // 1. If input is an instance of expectedType, return stringified input and abort these steps.
+    switch (expected_type) {
+    case TrustedTypeName::TrustedHTML:
+        if (input.has<GC::Root<TrustedHTML>>())
+            return input.get<GC::Root<TrustedHTML>>()->to_string();
```

Comment:
`Variant::get_pointer<T>()` can be used to combine the `has()` and `get()` calls: it'll return null if it doesn't hold a `T`. Variant operations can be slow so it's nice to reduce the number of times we access its value.